File: solutions/python/loggingframework/file_appender.py
from log_appender import LogAppender

class FileAppender(LogAppender):
    def __init__(self, file_path):
        self.file_path = file_path
    
    def append(self, log_message):
        with open(self.file_path, "a") as file:
            file.write(str(log_message) + "\n")

File: solutions/csharp/loggingframework/FileAppender.cs
using System;
using System.IO;

namespace LoggingFramework
{
    public class FileAppender : LogAppender
    {
        private readonly string _filePath;

        public FileAppender(string filePath)
        {
            _filePath = filePath;
        }

        public void Append(LogMessage logMessage)
        {
            try
            {
                using (var writer = new StreamWriter(_filePath, true))
                {
                    writer.WriteLine(logMessage.ToString());
                }
            }
            catch (IOException ex)
            {
                Console.WriteLine("FileAppender error: " + ex.Message);
            }
        }
    }
}